src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -126,9 +126,14 @@ public void SetDispatchMapIndex(int index)
             _optionalFieldsBuilder.SetFieldValue(EETypeOptionalFieldTag.DispatchMap, checked((uint)index));
         }
 
+        public static string GetMangledName(TypeDesc type, NameMangler nameMangler)
+        {
+            return "__EEType_" + nameMangler.GetMangledTypeName(type);
+        }
+
         public virtual void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         {
-            sb.Append("__EEType_").Append(nameMangler.GetMangledTypeName(_type));
+            sb.Append(GetMangledName(_type, nameMangler));
```

Comment:
This will allocate an extra temporary string. Undo this change. (It is fine to have the logic duplicated in the static method.)